sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -955,6 +957,18 @@ class Analyzer(
         Project(p.output, newPlan.withNewChildren(newChild :: Nil))
     }
   }
+
+  object CleanAlias extends Rule[LogicalPlan] {
```

Comment:
Maybe just search for `UnresolvedAlias` in this file and make sure we are consistent and only handle this in one place.